website/blog/2025-09-15-make-judge/index.mdx

```diff
@@ -0,0 +1,229 @@
+
+If you've tried evaluating GenAI applications, you know generic metrics miss what matters. Your customer support bot needs to be evaluated on empathy and problem resolution. Your code generator needs to be checked for security vulnerabilities. Your medical advisor needs domain-specific accuracy checks. MLflow Scorers let you define these requirements in plain English.
+
+This post demonstrates the power of MLflow Scorers by showing how to:
+
+- Create custom scorers with `make_judge` using natural language instructions
+- Build scorers that can act agentically, analyzing entire execution traces
+- Align scorers with human feedback to improve their accuracy over time
+- Share and version scorers across your team
+
+We'll build a customer support quality scorer, show how it can evaluate complex agent behaviors, and demonstrate how human feedback makes it smarter. The built-in scorers MLflow provides? They're just pre-defined versions using these same capabilities. Once you understand this, you'll see how flexible and powerful MLflow's evaluation framework really is.
+
+## Creating Your First Scorer: A Customer Support Evaluator
+
+Let's start with a practical example. Say you're building a customer support chatbot. You need a scorer that evaluates whether responses are actually helpful—not just grammatically correct.
+
+First, install MLflow:
+
+```shell
+pip install mlflow
+```
+
+Now let's create a scorer that understands what makes a good support response:
+
+```python
+from mlflow.genai.judges import make_judge
+
+# Create a scorer for customer support quality
+support_scorer = make_judge(
+    name="support_quality",
+    instructions=(
+        "Evaluate if the response in {{ outputs }} properly addresses "
+        "the customer issue in {{ inputs }}.\n\n"
+        "Consider: tone, helpfulness, clarity, and completeness.\n"
+        "Rate as: 'excellent', 'good', 'fair', or 'poor'"
+    ),
+    model="anthropic:/claude-3-opus-20240229"
+)
+```
+
+The key here is that we're defining a scorer using plain English. No complex scoring functions, no regex patterns—just describe what you're looking for. This scorer is now a reusable evaluation component that works anywhere in MLflow's ecosystem.
+
+Let's see how our scorer handles a terrible support response:
+
+```python
+# Test the scorer on a support interaction
+result = support_scorer(
+    inputs={"issue": "Can't reset my password"},
+    outputs={"response": "Have you tried turning it off and on again?"}
+)
+
+print(f"Rating: {result.value}")
+print(f"Reasoning: {result.rationale}")
+```
+
+Output:
+
+```
+Rating: poor
+Reasoning: The response does not address the password reset issue at all.
+It provides a generic IT troubleshooting suggestion that is irrelevant to
+the customer's specific problem...
+```
+
+### Scaling Up: Evaluating Entire Datasets
```

Comment:
This describes existing functionality. I would suggest just adding the code snippet in the previous section, with a note that `make_judge()` returns a Scorer than can be plugged in to `mlflow.genai.evaluate()` in order to scale up evaluations to entire datasets or in monitoring APIs to apply the scorers on production traffic.